test/unit/gateways/blue_snap_test.rb

```diff
@@ -543,6 +543,17 @@ def test_localizes_currencies
     end
   end
 
+  def test_optional_idempotency_key_header
+    response = stub_comms(@gateway, :raw_ssl_request) do
+      @gateway.purchase(@amount, @credit_card, @options.merge({ idempotency_key: 'test123' }))
+    end.check_request do |headers|
+      headers && headers['Idempotency-Key'] == 'test123'
+    end.respond_with(successful_authorize_response)
+
+    assert_success response
+    assert_equal '1012082893', response.authorization
```

Comment:
Nit: These two lines are superfluous, so you might as well remove them.

On line 551 you have stubbed the response, and here you are making assertions against the very response you provided.

I know this pattern has been copied and pasted throughout active merchant but I think the test is much nicer if you remove these lines because then it better communicates what you are actually trying to test (which is that the request itself contains the header).